packages/docusaurus/bin/beforeCli.mjs

```diff
@@ -105,18 +106,33 @@ export default async function beforeCli() {
       .map((p) => p.concat('@latest'))
       .join(' ');
 
+    const getYarnVersion = async () => {
+      if (!(await fs.pathExists(path.resolve('yarn.lock')))) {
+        return undefined;
+      }
+
+      const yarnVersionResult = shell.exec('yarn --version', {silent: true});
+      if (yarnVersionResult?.code === 0) {
+        const majorVersion = parseInt(
+          yarnVersionResult.stdout?.trim()?.split('.')[0] ?? '',
```

Comment:
```suggestion
          yarnVersionResult.stdout?.trim().split('.')[0] ?? '',
```

Reply:
Guess we don't need optional chaining there, cause nullish coalescing operator still provides the fallback value😊 Thanks!